jit-compiler/src/codegen.rs

```diff
@@ -315,6 +336,88 @@ impl<'a, T: FieldElement> CodeGenerator<'a, T> {
     }
 }
 
+/// Used for patterns in match and let statements:
+/// `value_name` is an expression string that is to be matched against `pattern`.
+/// Returns a rust pattern string (tuple of new variables, might be nested) and a code string
+/// that, when executed, returns an Option with the values for the new variables if the pattern
+/// matched `value_name` and `None` otherwise.
+///
+/// So if `let (vars, code) = check_pattern("x", pattern)?;`, then the return value
+/// can be used like this: `if let Some({vars}) = ({code}) {{ .. }}`
+fn check_pattern(value_name: &str, pattern: &Pattern) -> Result<(String, String), String> {
```

Comment:
idk about the name, maybe `match_arm_pattern`?

Reply:
It will also be used in `let` statements.